subsys/net/lib/prometheus/histogram.c

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2024 Mustafa Abdullah Kus, Sparse Technology
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/net/prometheus/histogram.h>
+
+#include <stdlib.h>
+#include <string.h>
+
+#include <zephyr/kernel.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(pm_histogram, CONFIG_PROMETHEUS_LOG_LEVEL);
+
+int prometheus_histogram_observe(struct prometheus_histogram *histogram, double value)
+{
+	if (!histogram) {
+		return -EINVAL;
+	}
+
+	/* increment count */
+	histogram->count++;
+
+	/* update sum */
+	histogram->sum += value;
+
+	/* find appropriate bucket */
+	for (size_t i = 0; i < histogram->num_buckets; ++i) {
+		if (value <= histogram->buckets[i].upper_bound) {
+			/* increment count for the bucket */
+			histogram->buckets[i].count++;
+			break;
```

Comment:
This means that values out of range will silently be ignored instead of triggering an error or land in the top bucket.

If you wished to avoid it, you could change `break;` by `return 0;` and at the bottom of the function set maybe a `return -ERANGE;` or `-EOVERFLOW` or `-ENOTSUP` or something more fitting.

No need to change that if that was the expected behavior though!
In which case, it is also possible to add a `LOG_DBG()` to help someone using the lib to understand "why is my value never added?".

Reply:
It is `Apdex score` related issue but you are right, user want to know and debug values in some cases. Added debug message for condition branch.

Reply:
Insightful, thank you for the reference and modification!